platform/mbed_board.c

```diff
@@ -55,36 +50,41 @@ void mbed_error_printf(const char *format, ...)
 {
     va_list arg;
     va_start(arg, format);
-    mbed_error_vfprintf(format, arg);
+    mbed_error_vprintf(format, arg);
     va_end(arg);
 }
 
-void mbed_error_vfprintf(const char *format, va_list arg)
+void mbed_error_vprintf(const char *format, va_list arg)
 {
-#if DEVICE_SERIAL
-#define ERROR_BUF_SIZE      (128)
     core_util_critical_section_enter();
-    char buffer[ERROR_BUF_SIZE];
-    int size = vsnprintf(buffer, ERROR_BUF_SIZE, format, arg);
-    if (size > 0) {
-        if (!stdio_uart_inited) {
-            serial_init(&stdio_uart, STDIO_UART_TX, STDIO_UART_RX);
-        }
-#if MBED_CONF_PLATFORM_STDIO_CONVERT_NEWLINES
-        char stdio_out_prev = '\0';
-        for (int i = 0; i < size; i++) {
-            if (buffer[i] == '\n' && stdio_out_prev != '\r') {
-                serial_putc(&stdio_uart, '\r');
-            }
-            serial_putc(&stdio_uart, buffer[i]);
-            stdio_out_prev = buffer[i];
+    char buffer[132];
```

Comment:
Could this buffer size be configurable? Now we have one rare MBED_ASSERT()ion failure in CI, which only logs part of the filename:
```mbed assertation failed: FALSE, file: ./mbed-os/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W�°��DY`—Ý<```
I suppose it would be better to have only the N trailing chars from the filename, but that might be a bigger change.

Reply:
This does partially address that assert issue; and both a commit I removed and #8255 address the other half.

This commit will make that be `mbed assertation failed: FALSE, file: ./mbed-os/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W...\n`

Which is an improvement - the `\n` means that you'll see the line in GreenTea, so you at least get the `FALSE`. (Moral of that story - use meaningful expressions in your asserts).

The commit I used to have here broke the assert up into 4 prints, so it was less likely to overflow and even if the filename was truncated you'd still see the line number.

#8255 does the same job by passing through to `mbed_error`, which also breaks up the prints.

Still, I can see there may be a desire for bigger buffer. Sadly, I can't see any way to get the "tail" of a printf.

Alternatively, how about avoiding the need for a big buffer by having `mbed_puts` instead, and using that for potentially-long things like the filename?

Reply:
Having a line number and non-garbage error print is definitely a step forward;). Could the filename trimming be done at mbed_assert_internal(), https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_assert.c#L25? 
That mbed_error_puts() would fix the problem completely, without need for trimming at all.

Reply:
Yes, mbed_assert_internal could use puts, or trim itself. But I've pulled back from any changes to that to defer to #8255. But as I said, that introduces a pile of new problems - or at least applies its existing problems to mbed_assert.